cmd/sovereignnode/dataCodec/dataCodec_test.go

```diff
@@ -0,0 +1,452 @@
+
+		deserialized, err := dataCodec.DeserializeEventData(eventData)
+		require.Nil(t, err)
+		require.Equal(t, expectedEventData, deserialized)
+	})
+}
+
+func TestDataCodec_SerializeTokenData(t *testing.T) {
+	t.Parallel()
+
+	dataCodec := createDataCodec()
+
+	addr0, _ := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+	amount := new(big.Int)
+	amount.SetString("123000000000000000000", 10)
+
+	esdtTokenData := sovereign.EsdtTokenData{
+		TokenType:  0,
+		Amount:     amount,
+		Frozen:     false,
+		Hash:       make([]byte, 0),
+		Name:       []byte(""),
+		Attributes: make([]byte, 0),
+		Creator:    addr0,
+		Royalties:  big.NewInt(0),
+		Uris:       make([][]byte, 0),
+	}
+
+	serialized, err := dataCodec.SerializeTokenData(esdtTokenData)
+	require.Nil(t, err)
+	require.Equal(t, "000000000906aaf7c8516d0c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", hex.EncodeToString(serialized))
+}
+
+func TestDataCodec_DeserializeTokenData(t *testing.T) {
+	t.Parallel()
+
+	dataCodec := createDataCodec()
+
+	t.Run("empty data should fail", func(t *testing.T) {
+		t.Parallel()
+
+		deserialized, err := dataCodec.DeserializeTokenData(nil)
+		require.Nil(t, deserialized)
+		require.Equal(t, errEmptyTokenData, err)
+	})
+	t.Run("token data should work", func(t *testing.T) {
+		t.Parallel()
+
+		addr0, _ := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+		amount := new(big.Int)
+		amount.SetString("123000000000000000000", 10)
+
+		expectedEsdtTokenData := &sovereign.EsdtTokenData{
+			TokenType:  0,
+			Amount:     amount,
+			Frozen:     false,
+			Hash:       make([]byte, 0),
+			Name:       []byte(""),
+			Attributes: make([]byte, 0),
+			Creator:    addr0,
+			Royalties:  big.NewInt(0),
+			Uris:       make([][]byte, 0),
+		}
+
+		tokenData, err := hex.DecodeString("000000000906aaf7c8516d0c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000")
+		require.Nil(t, err)
+
+		deserialized, err := dataCodec.DeserializeTokenData(tokenData)
+		require.Nil(t, err)
+		require.Equal(t, expectedEsdtTokenData, deserialized)
+	})
+}
+
+func TestDataCodec_GetTokenDataBytes(t *testing.T) {
+	t.Parallel()
+
+	dataCodec := createDataCodec()
+	marshaller := marshallerMock.MarshalizerMock{}
+
+	t.Run("empty token data should fail", func(t *testing.T) {
+		t.Parallel()
+
+		tokenDataBytes, err := dataCodec.GetTokenDataBytes(nil, nil)
+		require.Nil(t, tokenDataBytes)
+		require.Equal(t, errEmptyTokenData, err)
+	})
+	t.Run("fungible token should return only amount", func(t *testing.T) {
+		t.Parallel()
+
+		addr0, _ := hex.DecodeString("0000000000000000000000000000000000000000000000000000000000000000")
+		amount := new(big.Int)
+		amount.SetString("123000000000000000000", 10)
+		tokenNonce := make([]byte, 0)
+
+		esdtTokenData := sovereign.EsdtTokenData{
+			TokenType:  0,
```

Comment:
Even though this will be moved in another file, please use const for fungible here instead of 0, please